sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -98,8 +96,10 @@ private[sql] object DataSourceStrategy extends Strategy with Logging {
         l,
         projects,
         filters,
-        (a, f) =>
-          toCatalystRDD(l, a, t.buildScan(a.map(_.name).toArray, f, t.paths, confBroadcast))) :: Nil
+        (a, f) => {
+          t.refresh()
```

Comment:
After rethining about this, this can be removed, as the `InsertIntoHadoopFsRelation` will refresh the file status for us. we don't need to do that any more.